src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -270,6 +270,27 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             // Note: persist "m_currentThread" to be compatible with old versions
             info.AddValue(ThreadIDField, 0);
         }
+        
+        [System.Diagnostics.Conditional("DEBUG")]
```

Comment:
I will move this method to a new file since StringBuilder is already a partial class since this class is already long